### DISCUSSION TITLE
Cancel page build if API errors when using ISR

### DISCUSSION THREAD

**Discussion by elie222** [vercel/next.js#30017] [Help]
We have a situation where the API occasionally errors and returns no data (due to a db connection error).

In that scenario we want to cancel the new build and use the previous build instead so that the site still shows data instead of an empty table. Is there a way to do this with ISR?

**Comment by michaelschufi** [comment]
Have you tried throwing an error during `getStaticProps` if you are missing data? 

The idea being, that it can't complete `getStaticProps` and therefore won't use the new props and replace your HTML.

**Reply by elie222** [reply]
Thanks! Will try this... We had a try/catch set up so it could well have been this :(
